modules/zip/src/parse-zip/end-of-central-directory.ts

```diff
@@ -66,6 +68,7 @@ export const parseEoCDRecord = async (file: FileProvider): Promise<ZipEoCDRecord
   const zipEoCDOffset = await searchFromTheEnd(file, eoCDSignature);
 
   let cdRecordsNumber = BigInt(await file.getUint16(zipEoCDOffset + CD_RECORDS_NUMBER_OFFSET));
+  let cdByteSize = BigInt(await file.getUint32(zipEoCDOffset + CD_CD_BYTE_SIZE_OFFSET));
```

Comment:
You should read the full chunk first, just like above. Waiting for each couple of bytes is going to destroy performance.

Reply:
There's only one EoCD in every archive, so this code runs only once per slpk and doesn't affect performance at all in my observations. So I'll optimise this part in future PRs